HIP/hip-1056.md

```diff
@@ -0,0 +1,1952 @@
+a bit more complex as they would have to be aware of and handle this marker case.
+The final solution was to capture events created by signing Block `n` in the Pre Consensus Event Stream (PCES) across the upgrade
+boundary for insertion into a future Block `n+1` after upgrade.
+
+#### Impact of large entities when externalizing individual entity state. 
+The Block Stream plans to ensure visibilty of all modified entities by externalizing the full entity state in the stream.
+This is to solve the issue of partial data in record streams where a client may miss the externalization of state data and would
+therefore possess a partial state for entities with few easy options to retrieve the missing data.
+By externalizing the full entity state in the block stream a client can get the complete entity details on any update and won't have
+to employ "upsert" logic. An emergent feature when externalizing the full entity state of modified entities is that when entity
+properties are modified multiple times those entities may be externalized in the stream multiple times.
+These accounts may have multiple keys and other potentially large list properties (e.g. approvals) that may result in a large block size.
+To manage this an idea inspired from video iframe approach (when the image changes the whole entity is sent, if not only diffs
+are sent) was considered. In this case every entity could have a last updated counter that is updated on entity changes, if the
+value is over a certain threshold then the whole entity is externalized, if not only deltas would be externalized. This would optimize
+how far back a client would have to go to retrieve state info. However, this approach would require the counter to be stored in state
+for every entity which has undesired impacts on state size.
+Another inspired approach was that on the first observation of an entity in a block the CN would externalize the full entity state,
+after that only the delta changes would be externalized for the remainder of the block.
+In this way any consumer of a block would have the complete state by applying the diffs onto the initial state.
+This shares the downside of requiring additional temporary storage, and also requires the creation of a "difference" form for every
+entity potentially stored in state, and could, paradoxically, _increase_ the total size of blocks with very many changes to small entities.
+
+### Including exchange_rate in the TransactionResult
+In the past the exchange_rate details were in the transaction record. However, this information doesn't change often and is based on
+the value of a system file. Changes to this file are externalized to the MN and thus the current exchange rate value is always available
+via a MN (or future BN). Thus the exchange rate is readily available from the MN when using the transaction timestamp as input.
+This allowed us to removing it and reduce unnecessary data from the block stream.
```

Comment:
```suggestion
In the past the exchange_rate details were in the transaction record.
However, this information doesn't change often and is based on the value
of a system file. Changes to this file are also externalized in the
Block Stream, thus the current exchange rate value is available in the state
changes and we can remove it to reduce unnecessary data in the block stream.
```

Reply:
The move from 120 to 80 is just my own habit because it makes github "suggestions" easy to read; I do not know of any requirement that you stick to it or reformat the entire file to match.

I removed the note about MN because it does not justify removing the value from the BlockStream.  If the exchange rate details are not _somewhere_ in the BlockStream, then MN _will not_ have the value. MN has to get the value _from_ the BlockStream, so claiming we can remove the data because it's available in MN is somewhat circular, in my opinion.